src/App.tsx

```diff
@@ -260,12 +260,9 @@ function App() {
                   }
                   onClear={(clearedIndex) => {
                     setValues((values) => {
-                      return values.map((value) => {
-                        if (index === clearedIndex) {
-                          return {value: ''}
-                        }
-                        return value
-                      })
+                      const newValues = [...values]
+                      newValues[clearedIndex] = {value: ''}
+                      return newValues
```

Comment:
It seems that both the old and the new code are doing the same thing. Are you sure this is what was causing the bug?

Reply:
Oh god you're right, I just verified and it should have been:

```
  return values.map((value, innerIndex) => {
      if (innerIndex === clearedIndex) {
        return {value: ''}
      }
      return value
    })
```

I got confused with the outer index available 🤦